// Copyright 2019 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Forked from mediapipe/framework/status_handler.proto.
// The forked proto must remain identical to the original proto and should be
// ONLY used by mediapipe open source project.
//
// Messages for specifying StatusHandlers.

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/mediapipe_options.proto";

// The settings specifying a status handler and its required external inputs.
message StatusHandlerConfig {
  // The name of the registered status handler class.
  optional string status_handler = 1;  // required

  // The name of the input side packets.  The StatusHandler can access its
  // input side packets by index or by tag.  A StatusHandler will only
  // be called if all of its requested input side packets are available
  // (and won't be called if a PacketFactory or PacketGenerator which
  // produces one fails).
  repeated string input_side_packet = 2;
  // DEPRECATED(mgeorg) The old name for input_side_packet.
  repeated string external_input = 1002;

  // The options for the status handler.
  optional MediaPipeOptions options = 3;
}
